app/src/main/java/org/simple/clinic/widgets/qrcodescanner/MLKitQrCodeAnalyzer.kt

```diff
@@ -0,0 +1,43 @@
+package org.simple.clinic.widgets.qrcodescanner
+
+import android.annotation.SuppressLint
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageProxy
+import com.google.mlkit.vision.barcode.Barcode
+import com.google.mlkit.vision.barcode.BarcodeScannerOptions
+import com.google.mlkit.vision.barcode.BarcodeScanning
+import com.google.mlkit.vision.common.InputImage
+
+class MLKitQrCodeAnalyzer(
+    private val onQrCodeDetected: OnQrCodeDetected
+) : ImageAnalysis.Analyzer {
+
+  private val options = BarcodeScannerOptions.Builder()
+      .setBarcodeFormats(Barcode.FORMAT_QR_CODE)
+      .build()
+  private val scanner = BarcodeScanning.getClient(options)
+
+
+  @SuppressLint("UnsafeExperimentalUsageError")
+  override fun analyze(imageProxy: ImageProxy) {
+    val mediaImage = imageProxy.image
+    if (mediaImage != null) {
+      val image = InputImage.fromMediaImage(mediaImage, imageProxy.imageInfo.rotationDegrees)
+      scanner.process(image)
+          .addOnSuccessListener { barcodes ->
```

Comment:
"Each Barcode object represents a barcode that was detected in the image." Does this mean that the list of barcodes is the multiple images that are taken while scanning?

Reply:
No, it represents multiple barcodes detected on a single scan.